src/fileformats/ocd_types.h

```diff
@@ -683,7 +704,8 @@ OcdEntityIndexIterator<V> OcdEntityIndexIterator<V>::operator++(int)
 template< class V >
 typename OcdEntityIndexIterator<V>::value_type OcdEntityIndexIterator<V>::operator*() const
 {
-	return { &block->entries[index], reinterpret_cast<const typename value_type::EntityType*>(byte_array->data()+block->entries[index].pos) };
+	auto entry_ptr = Ocd::getPointerChecked< const typename value_type::EntityType >(*byte_array, block->entries[index].pos);
+	return { &block->entries[index], entry_ptr };
```

Comment:
(1) This validation is performed each time the iterator is dereferenced, and it requires checking the result in calling code. However I intended to make non-dereferencable entries unreachable via iterators, i.e. ATM you cannot reach entries with `pos` zero.

https://github.com/lpechacek/mapper/blob/ebcc8837f596d5b50e7e06bc52e4b048030e0b57/src/fileformats/ocd_types.h#L668-L694

So I wonder if it would be possible to move the pointer validation to `OcdEntityIndexIterator` constructor and `operator++`, thus making entries with invalid pointer unreachable. However I can't tell now if this would work for strings, symbols and objects.